### PR TITLE
Update Python Software Foundation Copyright Year.

### PR DESCRIPTION
**Pull request by orsenthil**
Make it current.

I searched for `\d+ Python Software Foundation.` and selectively updated where it made sense.

### CODE REVIEW

**Review by Mariatta — APPROVED** [review]

File: Lib/_pydecimal.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2004 Python Software Foundation.
+# Copyright (c) 2017 Python Software Foundation.
```
Comment (willingc) [inline Lib/_pydecimal.py]: Perhaps on these where the source was a single year, it should convert to `2004-2017`

**Review by willingc — COMMENTED** [review]
Might wish to look at the conversion from a single date.

File: Lib/_pydecimal.py (hunk 1 above)
Comment (Carreau) [reply Lib/_pydecimal.py]: Usually `Copyright (c) <year>` means `<year>-present`, IMHO this should stay  2004.  and 2017 -> Present.

This avoid churn on file and to keep them updated.

**Comment by gpshead** [comment]
I strongly recommend NOT doing this.

I believe this has come up before, do not edit the date or otherwise turn it into a sequence or range of dates in a copyright notice in existing files. Leave the date in the file as it was originally written. It is meaningless churn at best.

**Comment by Carreau** [comment]
http://www.copyrightlaws.com/copyright-basics/copyright-notice-year/ :

> The general rule is that the year to include in a copyright notice is **the year of first publication** of the work 

> the notice **may** include a range of years (e.g., 2009-2013), starting from the date of the oldest published elements and ending with the date of the newest published elements.

Emphasis mine. Also the last year should be changed only if the content of the file have changed, so I would also favor the 1 year entry only.

**Comment by orsenthil** [comment]
@gpshead - you mean no to the entire change? I know this has come up earlier.

How about removing the copyright line from the modules headers altogether (when appropriate) ? It is mentioned in the README.

* For ranges which are wrong like 2001-2007 Python Software Foundation. There is no controversy, we should either remove it or update it.

* For range copyright start year. As others have pointed out, leaving it start year seems like a good idea.

File: Lib/_pydecimal.py (hunk 1 above)
Comment (brettcannon) [reply Lib/_pydecimal.py]: Maybe change this PR to strip the date range from the first one and leave the other alone? @VanL can you weigh in?

**Comment by zware** [comment]
@benjaminp, you have historically done the copyright update, what criteria have you used?

**Comment by warsaw** [comment]
Here's what I use to bump my copyright years.  Feel free to beg, borrow, or steal.

https://gitlab.com/mailman/mailman/blob/master/copybump.py

**Comment by benjaminp** [comment]
I generally update the ones that cover "all of Python" like the LICENSE and `getcopyright.c`. My preference would be removing all PSF copyright headers from internal files and rely on the repository-level ones. (Ones for contributed-and-licensed code should just be left alone.)

**Comment by malemburg** [comment]
Please don't remove the copyright notices from the files. The origin of the code and copyright status is already hard to determine given Python's history. Removing the notices would make this even harder.

**Comment by Alir3z4** [comment]
`range(2001, 2018)` would be good or `xrange(2001, 2018)` or even `xrange(2001, CURRENT)`

**Review by malemburg — CHANGES_REQUESTED** [review]
The change in Mac/BuildScript/resources/License.rtf is wrong (you removed 2016). Please also verify other license documents you may have touched.

**Comment by benjaminp** [comment]
I do not believe copyright notices referring to anything but the PSF
should be removed.

**Comment by orsenthil** [comment]
Acknowledge. I was suggesting we remove only the PSF Copyright lines in the internal source modules. The author ones will remain in tact.

**Comment by malemburg** [comment]
Just like with all copyright notices, removals of PSF copyright notices is something only the PSF board can decide. 

IMO, it would be better to find a short PSF notice text (e.g. one without year, so that we don't have to touch the files once every year) and add it to all files which currently do not have it, just like GNU projects do. It makes tracking copyrights much easier.

**Comment by benjaminp** [comment]
> Just like with all copyright notices, removals of PSF copyright notices is something only the PSF board can decide.

We should ask them then.

> 
> IMO, it would be better to find a short PSF notice text (e.g. one without year, so that we don't have to touch the files once every year) and add it to all files which currently do not have it, just like GNU projects do. It makes tracking copyrights much easier.

Not having copyright headers distributed around the entire sounds easiest to me. It's also not clear to me that adding a copyright header to every cpython source file would be correct, since contributing to Python does not require copyright assignment.

**Comment by malemburg** [comment]
On 13.02.2017 08:25, Benjamin Peterson wrote:
>> IMO, it would be better to find a short PSF notice text (e.g. one without year, so that we don't have to touch the files once every year) and add it to all files which currently do not have it, just like GNU projects do. It makes tracking copyrights much easier.
> 
> Not having copyright headers distributed around the entire sounds easiest to me. It's also not clear to me that adding a copyright header to every cpython source file would be correct, since contributing to Python does not require copyright assignment.

It would be correct for all files where the PSF does have the copyright
(we do have copyright assignments for quite a bit of code as well),
but I see your point. Trying to do this correctly will be a project
on its own.

-- 
Marc-Andre Lemburg
http://www.malemburg.com/

**Comment by nedbat** [comment]
For the main README file: the long list of years is in the file twice.  Surely we don't need to state it twice.  Let's get rid of the one at the top of the file, which is just clutter preventing people from reading the file.

**Comment by orsenthil** [comment]
I have addressed the review comments.

* Aim is to be non-controversial.

File: Lib/lib2to3/pgen2/tokenize.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006 Python Software Foundation.
+# Copyright (c) 2001, 2002, 2003, 2004, 2005, 2017 Python Software Foundation.
```
Comment (malemburg) [inline Lib/lib2to3/pgen2/tokenize.py]: This change is wrong. You should either add all years, or leave things as they are.

**Comment by malemburg** [comment]
Ok, so following on the discussion on our committers list, I'd like to suggest the following:

 * you open up a ticket on bpo where we can discuss changes
 * this PR is closed, since there are too many controversial changes; in particular, doing a simple grep&replace is the not way we can address this -- copyright notices are legal marks in the code, so special care has to be taken not to break them in ways which change their legal meaning
 * a new PR is opened once we have agreed on a way forward

Overall, we should stop having non code review related discussions on PRs.

**Comment by orsenthil** [comment]
@malemburg , I think, you commented on an old diff or reviewed the outdated patch. (See the comment folded under Show Outdated). The latest diff can see by clicking on files changed again: https://github.com/python/cpython/pull/4/files

I will move this discussion to bpo.

Given our to move to github, I suspect, the contributions and reviews are going to happen in GIthub PR than bpo. One way to look at it this, bpo is place where you discuss and decide upon something. Github PR is the place where you chalk out details or fine tune it before committing.

**Comment by orsenthil** [comment]
Here is the bpo issue: http://bugs.python.org/issue29555

Let's decide the future state for the codebase in that link.

**Pull request closed without merging**